contracts/Payments.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@api3/airnode-protocol-v1/contracts/dapis/DapiReader.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+contract Payments is ERC721, DapiReader, Ownable {
+    using Counters for Counters.Counter;
+
+    Counters.Counter private _tokenIdCounter;
+
+    ERC20 public _ERC20;
+    mapping(address => bytes32) public tokenDapiMapping;
+    mapping(uint256 => uint256) public TokenIDtoPrice; 
+
+constructor(address _dapiServer, address _ERC20Address) DapiReader(_dapiServer) ERC721("Payment Receipt", "PRT") {
+        _ERC20 = ERC20(_ERC20Address);
+    }
+
+    function setDapiName(address token, bytes32 DapiName)
+        public
+        onlyOwner
+    {
+        tokenDapiMapping[token] = DapiName;
+    }
+
+    function getTokenPrice(address token) public view returns (uint256 tokenPriceUint256) {
+        bytes32 DapiName = tokenDapiMapping[token];
+        int224 value = IDapiServer(dapiServer).readDataFeedValueWithDapiName(
+                DapiName
+        );
+        uint224 tokenPriceUint224 = uint224(value);
```

Comment:
typecasting it to `uint224` first

Reply:
> The task is pretty simple, you need to
> 
>     1. Accept payments in multilple types of token
> 
>     2. Get their USD value
> 
>     3. Mint an NFT with the USD value
> 
> 
> The tests also aren't done properly because they are actually not testing anything (payment is made? but your not even checking if the `Dapiserver` is converting from the token value to usd value). Rewrite the contract and please pay attention to what is required and test it out yourself with the intended usage. The contract is not according to specs.

- Fixed all the issues, contract can now accept multiple ERC20 token payments. Added a function to add allowed tokens to be added. 

- Added another token, Mock Wrapped Matic.

- `ownerWithdrawFunds` will now withdraw all `allowedTokens` held by the smart contract. 

- `getTotalPayments` will now return total USD balance of all the allowed ERC20 tokens held by the smart contract.

- Added proper comments for each method.

- Didn't get the part where you said the `Payment` function isn't checking if the value is converted by calling the dAPI. The function will just throw an error: `Sender cannot read` if it's not able to read the dAPI. Also, added more error handling for it to be safer.

Reply:
its already a uint224

Reply:
```tokenPriceUint256 = uint256(uint224(value));```